php_phongo.c

```diff
@@ -2096,23 +2096,23 @@ static mongoc_ssl_opt_t* php_phongo_make_ssl_opt(mongoc_uri_t* uri, zval* zoptio
 static void php_phongo_free_ssl_opt(mongoc_ssl_opt_t* ssl_opt)
 {
 	if (ssl_opt->pem_file) {
-		str_efree(ssl_opt->pem_file);
+		efree((char*) ssl_opt->pem_file);
```

Comment:
Is `(char*)` necessary? I noticed you didn't do it for `efree(key_vault_ns)` later on in this file.

`efree` itself takes a [`void*`](http://lxr.jpauli.tech/source/xref/php-7.4/Zend/zend_alloc.c#2541), just like `free()`, so any pointer should suffice without a cast.

Reply:
```
php_phongo.c:2083:3: error: passing 'const char *' to parameter of type 'void *' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
                efree(ssl_opt->pem_file);
                ^~~~~~~~~~~~~~~~~~~~~~~~
```

`pem_file` is declared as a `const char*` in `_mongoc_ssl_opt_t`, along with other string options. We create the `ssl_opt` member, its settings are later copied to mongoc, then we free it. The explicit cast makes sense, we just hid it behind a macro before.